### ISSUE TITLE
Rethink the UI and settings page

### ISSUE THREAD

**Issue by robinpaulson** [martykan/forecastie#397]
We are adding more information to the app; we should consider how to display it, and if we allow users to turn on/off certain (or all) pieces of information. There are three areas to consider.
1. weather now
2. weather forecast
3. charts
Perhaps we offer the same info for all three areas, with the ability to turn it off (I've left out sunset/sunrise, as I've decided we will only show it for today).
Info:
Wind speed
Wind direction
Temperature
Rain/snow
Humidity
Pressure
Wordy description (e.g. cloudy, light rain, clear)
UV
CO
NO2
SO2
Ozone

**Comment by robinpaulson** [comment]
The obvious way to do this would be to have a table in the settings, down the left-hand side we could have all the information i listed above (rain amount, wind direction, O2 pollution) and across the top the three places we could show this information (conditions now, forecast conditions, graphs).

**Comment by robinpaulson** [comment]
Something like this
![settings](https://user-images.githubusercontent.com/1186107/69602289-ee1b2680-107b-11ea-81d4-94a993924432.png)

**Comment by robinpaulson** [comment]
I'm going to use this issue to collect all the "please add weather info [x] to the forecast/graphs", hence i have closed a set of issues which individually requested that info.

**Comment by robinpaulson** [comment]
Given that most people will only be interested in a particular subset, we could hide the matrix behind an "Advanced" button in the settings. Then we set the default view to the most likely things they will need (maybe temperature, wind strength, rain, aggregated pollution, UV) as default.

**Comment by FridoDeluxe** [comment]
> Given that most people will only be interested in a particular small set of information, we could hide the matrix behind an "Advanced" button in the settings. Then we set the default view to the most likely things they will need (maybe temperature, wind strength, rain, aggregated pollution, UV) as default.

Yes.

Regarding how to present the options: Currently a lot of these options are not (yet) implemented and things like wordy forecast are for example not applicable to graphs. So there are differences between the "Now" / "Forecast" / "Graphs" sections and I would propose to split it into three different options. It makes them less complicated and they can be added to the current Settings Activity easily via a `<MultiSelectListPreference>` (behaves similar to the other pop-up options). This adds just three new rows to the list of settings and looks like this:

![forecastie-options](https://user-images.githubusercontent.com/498537/69761248-1a5fa180-1167-11ea-9175-28033abceb99.png)

Another idea or rather a question: Should the user be able to sort the options? Theoretically this would be nice, but I'm not sure how this can be realized (how UI should look like and how it's implemented).

**Comment by robinpaulson** [comment]
I agree with the first part about splitting into three, good point. Nice mock up too.

No to sorting for the moment, but i'm sure someone will ask for it at some point.

**Comment by robinpaulson** [comment]
Perhaps we need a new dev branch, to test these changes, with releases so users can help with feedback.

**Comment by robinpaulson** [comment]
Do we need similar for the widget?

**Comment by FridoDeluxe** [comment]
> No to sorting for the moment.

Same for me, also makes things a lot easier.

> Perhaps we need a new dev branch, to test these changes, with releases so users can help with feedback.

I don't know how easy it is to set this up (only a new branch and home-made APKs or is beta-distribution possible via F-Droid?). Just adding these three settings options and hiding/showing the values depeding on the preferences shouldn't be that complicated.

> Do we need similar for the widget, the ability to select which info is shown? Maybe we only need one widget then, which is complete configurable.

Making all the widgets consistent or configurable is definetly a good idea. But I think the widgets still differ a lot from the app's activities (for example in layout) so the mechanisms are different and it's worth an own issue.

**Comment by robinpaulson** [comment]
Now I think more about it, I think you're right on the second part, we don't need a dev branch for these changes (as an aside, f-droid handles beta/dev branches nicely).

As for widgets, we should probably put that in a second issue, it is different enough to the app UI.

**Comment by Currrupted** [comment]
> 
> 
> > Given that most people will only be interested in a particular small set of information, we could hide the matrix behind an "Advanced" button in the settings. Then we set the default view to the most likely things they will need (maybe temperature, wind strength, rain, aggregated pollution, UV) as default.
> 
> Yes.
> 
> Regarding how to present the options: Currently a lot of these options are not (yet) implemented and things like wordy forecast are for example not applicable to graphs. So there are differences between the "Now" / "Forecast" / "Graphs" sections and I would propose to split it into three different options. It makes them less complicated and they can be added to the current Settings Activity easily via a `<MultiSelectListPreference>` (behaves similar to the other pop-up options). This adds just three new rows to the list of settings and looks like this:
> 
> ![forecastie-options](https://user-images.githubusercontent.com/498537/69761248-1a5fa180-1167-11ea-9175-28033abceb99.png)
> 
> Another idea or rather a question: Should the user be able to sort the options? Theoretically this would be nice, but I'm not sure how this can be realized (how UI should look like and how it's implemented).

Perhaps it would be useful if the users could easily "add" or manually append their desired values to their customised list. I am sorry in case this is what you had planned anyway, but maybe this could be achieved by using a dialogue in which the user has their own "home screen", where they could then customize the specific elements that they need or want to display.

I was thinking of something similar to this, in the popular application OsmAnd for Android:
![grafik](https://user-images.githubusercontent.com/13292623/70044143-092cef80-15c2-11ea-9669-3e829fd058b8.png)
Customization of this certain page could be possible by dragging elements into the screen, or simply by offering a list of option which can be en-/disabled by hand.

**Comment by robinpaulson** [comment]
@Currrupted Yes, I think that's broadly what we have been talking through. As to the drag-and-drop, I'd rather keep it simple though, implementing that sort of UI sounds difficult and unnecessary for the moment.

**Comment by FridoDeluxe** [comment]
> ... simply by offering a list of option which can be en-/disabled by hand.

I think is exactly what I proposed with the screenshot.

Unfortunately there is not much time in the next weeks, so I can probably start only in the new year working on this. Until then there are maybe more comments and other/better ideas for this feature.

**Comment by robinpaulson** [comment]
As with everything here, we get to it when we can. Any work you do at any time is appreciated.

**Comment by robinpaulson** [comment]
Another coder proposed a PR today, #414, which adds cloud maps to Forecastie. This of course raises the question of which extra maps we might or might not include in the app. Anecdotally I get the impression less people use them than the graphs and text forecasts, but it it worthwhile researching which map tiles are available and considering which if any we add. I'll do no more than add this comment for now, but we should perhaps revisit it once we've settled on changes to the other elements.

**Comment by MisterY** [comment]
Speaking of maps, I would highly suggest https://earth.nullschool.net/, which I've seen in another weather app. Would you like to track the request in a separate issue or is here ok?

**Comment by qwertyuser1983** [comment]
![image](https://user-images.githubusercontent.com/22538918/91943781-5394f580-ed27-11ea-8da6-178857ccac56.png)

I really need the Feels Like temperature. In this thread, they are called the Realfeel, but afaik, that is Accuweather patent

**Comment by robinpaulson** [comment]
> Speaking of maps, I would highly suggest https://earth.nullschool.net/, which I've seen in another weather app. Would you like to track the request in a separate issue or is here ok?

Sorry, i missed this until now. I'm sure there are useful maps there, I think this is best captured under the existing issues which cover other weather providers than OSM. #126 for example.

**Comment by robinpaulson** [comment]
> I really need the Feels Like temperature.

I'm not sure what you mean?

There are multiple features requests and bugs in Forecastie. The issue is not whether thing X is more or less important than thing Y, but whether or not there is anyone willing and able to fix it. That determines the majority of the direction of this project. If you're prepared to code any of the things listed on the issues page, go for it.

**Comment by qwertyuser1983** [comment]
@robinpaulson I refer to your first post, https://github.com/martykan/forecastie/issues/397#issue-525859987

My bad, I may have formulated my post wrong. 
 
What you have described as  "Real Feel Temperature" is called "Feels Like Temperature" in OpenWeatherMap, possibly to avoid Accuweather Copyright Infringement. 

And yes, for me, this is the top priority, so thanks for the permission. 

(´｡• ω •｡`)

**Comment by robinpaulson** [comment]
> What you have described as "Real Feel Temperature" is called "Feels Like Temperature" in OpenWeatherMap, possibly to avoid Accuweather Copyright Infringement.

OK, first post updated to reflect this info

**Comment by huuhaa** [comment]
Great to see Feels Like Temperature to be added, when will it appear at the app ( or am I just with wooden eye)? And is there any chance to include some factor which depend of sun angle and effects to feels like temperature, or is that once again too similar to Accuweather patented method?

Other than that, I think it would be great to be able to set notification if airpressure drops x value / 3h, similar to Garmin watches storm alarm function which is adjustable.

**Comment by robinpaulson** [comment]
> Other than that, I think it would be great to be able to set notification if airpressure drops x value / 3h

I think this has been covered elsewhere, perhaps in #192
